cosign/README.md

```diff
@@ -0,0 +1,79 @@
+# Cosign Verifier
+
+## Introduction
+
+This package contains the source code for the ratify-verifier-go cosign client library.
+
+This package is built on the `sigstore-go` library, a low-level Go library that provides a suite of tools and standards designed for signing, verifying, and securing software supply chains.
+`sigstore-go` focuses on the core functionalities of signing and verification, offering a minimal and user-friendly API without integration of registry operations or additional features like container image management.
+In contrast, Cosign contains higher-level APIs within the Sigstore ecosystem that integrates these functionalities with container registries and other tools for end-to-end supply chain security.
+Cosign builds on `sigstore-go` and extends its capabilities to include container image signing, verification, and storage in registries.
+
+Using `sigstore-go` library makes it a more streamlined choice for developers seeking to implement lightweight and flexible signature verification.
+Additionally, the Sigstore team plans to refactor parts of Cosign to rely more heavily on `sigstore-go`, making the interaction between the two libraries even closer.
```

Comment:
The introduction is a sort of literature review. It must be logically structured.

- What is cosign? How is it related to SigStore (background)
- What's special about cosign? (motivation)
- Why should we use cosign? (motivation)
- How is cosign built? --> `sigstore-go` (analysis / decomposing)
- This verifier implementation uses the same library as for building cosign (proposal)

Reply:
By sharing these thoughts this introduction can be extremely long. Shall we explain that in a blog? 

cc\ @shizhMSFT @binbin-li

Reply:
See your point there. Going to update accordingly.

Reply:
Update.